### PR TITLE
Feature: Forgot password flow

### PR DESCRIPTION
**Pull request by mariusjagminas**

### CODE REVIEW

File: src/email-verification/emailVerification.templates.ts [hunk 1]
```diff
@@ -35,3 +35,37 @@ export const emailVerificationBody = (email: string, code: string): string => {
...
+  const tokenUri = encodeURI(token);
+  return `
+    <p>You are receiving this email because we received a password reset request for your account.</p>
+    <table width="100%" cellspacing="0" cellpadding="0">
+     <tr>
+        <td>
+            <table cellspacing="0" cellpadding="0">
+                <tr>
+                    <td style="border-radius: 24px;" bgcolor="#ef9c1b">
+                        <a href="${webAppBaseUrl}/reset-password?token=${tokenUri}&id=${id}"
```
Comment (kowalski) [inline src/email-verification/emailVerification.templates.ts:49]: Don't use query string, instead put the params into path. 

This is important for ios guys. Through deeplinking they can get params from path, but not from query string.

File: src/users/users.service.ts [hunk 2]
```diff
@@ -202,4 +209,55 @@ export class UsersService {
...
+
+      await this.emailService.emailUser(
+        user.email,
+        passwordResetSubject,
+        resetPasswordBody(token, user.id),
+      );
+
+      return {};
+    } catch (error) {
+      throw new NotFoundException(MSG.EMAIL_NOT_FOUND);
```
Comment (kowalski) [inline src/users/users.service.ts:235]: You are assuming here that all the possible errors are caused of wrong email. That's a bit of  a stretch. I can think different reasons for this to fail, yet here we are swallowing the original error and just returning 404

**Review by kowalski — CHANGES_REQUESTED** [review]

File: src/users/users.service.ts (hunk 2 above)
Comment (mariusjagminas) [reply src/users/users.service.ts:235]: I did that intentionally for security reasons. I think that here we should return only the generic message, without any explicit explanation what happened.?

File: src/users/users.service.ts [hunk 3]
```diff
@@ -232,7 +232,7 @@ export class UsersService {
 
       return {};
     } catch (error) {
-      throw new NotFoundException(MSG.EMAIL_NOT_FOUND);
+      throw new BadRequestException();
```
Comment (kowalski) [inline src/users/users.service.ts:235]: You are swallowing original error. At least log it. Swallowing errors leads to long unhappy hours of debugging in future

**Review by kowalski — APPROVED** [review]

**Pull request merged**